### DISCUSSION TITLE
How to avoid cron execution if previous still running?

### DISCUSSION THREAD

**Discussion by sercba** [psy0rz/zfs_autobackup#158] [Q&A] [answered]
Hello!

I've this cronjob:

    22 * * * * zfs-autobackup failover rpool/backup -v --debug --no-snapshot --clear-refreservation --compress --buffer 128M --strip-path 1 --ssh-source source --keep-source=4 --keep-target=4,1h1d,1d1w,1w1m

Do you know how to avoid a new cron execution if previous still running?

Thanks!

**Comment by Scrin** [comment, marked as answer]
You could use [flock](https://man7.org/linux/man-pages/man1/flock.1.html) for example. Something like this should do:
```
22 * * * * flock -n /var/backups.lock zfs-autobackup failover rpool/backup -v --debug --no-snapshot --clear-refreservation --compress --buffer 128M --strip-path 1 --ssh-source source --keep-source=4 --keep-target=4,1h1d,1d1w,1w1m

```

**Reply by sercba** [reply]
Thanks! I'll try that.
